backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmDeclarations.kt

```diff
@@ -262,35 +263,113 @@ private class DeclarationsGeneratorVisitor(override val context: Context) :
     }
 
     private fun createSingletonDeclarations(irClass: IrClass): SingletonLlvmDeclarations? {
-
         if (irClass.isUnit()) {
             return null
         }
 
         val isExported = irClass.isExported()
-        val symbolName = if (isExported) {
-            irClass.objectInstanceFieldSymbolName
+        val valueGetterName = if (isExported) {
+            irClass.objectInstanceGetterSymbolName
         } else {
-            "kobjref:" + qualifyInternalName(irClass)
+            null
         }
-        val threadLocal = irClass.storageKind(context) == ObjectStorageKind.THREAD_LOCAL
-        val instanceFieldRef = addGlobal(
-                symbolName, getLLVMType(irClass.defaultType), isExported = isExported, threadLocal = threadLocal)
-
-        val instanceShadowFieldRef =
-                if (threadLocal) null
-                else {
-                    val shadowSymbolName = if (isExported) {
-                        irClass.objectInstanceShadowFieldSymbolName
-                    } else {
-                        "kshadowobjref:" + qualifyInternalName(irClass)
+        val addressGetterName = if (isExported) {
+            irClass.objectInstanceAddressSymbolName
+        } else {
+            null
+        }
+
+        // If object is imported - access it via getter functions.
+        if (isExternal(irClass)) {
+            assert(valueGetterName != null)
```

Comment:
Is this code reachable?

Reply:
Note that `LlvmDeclarations` is explicitly designed to generate declarations for IR to be compiled. It is not supposed to cover any "external" IR declarations.